devito/function.py

```diff
@@ -362,30 +365,33 @@ def __init__(self, *args, **kwargs):
         if not self._cached():
             super(TimeFunction, self).__init__(*args, **kwargs)
             self.time_dim = kwargs.get('time_dim', None)
+            if self.time_dim is not None and not isinstance(self.time_dim, TimeDimension):
+                raise ValueError("time_dim must be a TimeDimension, not %s" %
+                                 type(self.time_dim))
             self.time_order = kwargs.get('time_order', 1)
-            self.save = kwargs.get('save', False)
-
-            if not self.save:
-                if self.time_dim is not None:
-                    warning('Explicit time dimension size (time_dim) found for '
-                            'TimeFunction symbol %s, despite \nusing a stepping time '
-                            'dimension (save=False). This value will be ignored!'
-                            % self.name)
-                self.time_dim = self.time_order + 1
-                self.indices[0].modulo = self.time_dim
+            self.save = kwargs.get('save', None)
+
+            if self.save is not None:
+                if not isinstance(self.save, int):
+                    raise ValueError("save must be an int indicating the number of " +
```

Comment:
Can't we instead fall back to full time size instead?

Reply:
This is the "full time size". Think of this as what `time_dim` was doing before - I am asking for the size of the time dimension here.